test/NuGet.Core.Tests/NuGet.Protocol.Tests/LocalResourceTests/LocalPackageSearchResourceTests.cs

```diff
@@ -597,5 +597,108 @@ public async Task LocalPackageSearch_SearchAsync_WithCancellationToken_ThrowsAsy
                     async () => await resource.SearchAsync("", null, 0, 1, NullLogger.Instance, new CancellationToken(canceled: true)));
             }
         }
+
+        [Theory]
```

Comment:
These tests are great, and probably sufficient the main tests that should be as the bug is *in* the resource, not the list command.